scripts/ai/AIDriveStrategyChopperCourse.lua

```diff
@@ -0,0 +1,782 @@
+
+function AIDriveStrategyChopperCourse:update(dt)
+    AIDriveStrategyCombineCourse.update(self, dt)
+    -- Old Code need to make Choppers work
+    self:updateChopperFillType()
+end
+
+function AIDriveStrategyChopperCourse:getDriveData(dt, vX, vY, vZ)
+    self:handlePipe()
+
+    if self.temporaryHold:get() then
+        self:setMaxSpeed(0)
+    end
+    if self.state == self.states.WORKING then
+        -- Harvesting
+        self:checkRendezvous()
+        self:checkBlockingUnloader()
+
+        if self:isChopperWaitingForUnloader() then
+            self:debug('No trailer calling for an unloader')
+            self:stopForUnload(self.states.WAITING_FOR_UNLOAD_ON_FIELD, true)
+        end
+
+    elseif self.state == self.states.TURNING then
+        self:checkBlockingUnloader()
+    elseif self.state == self.states.WAITING_FOR_LOWER then
+        if self:isChopperWaitingForUnloader() then
+            self:debug('No trailer calling for an unloader')
+            self:stopForUnload(self.states.WAITING_FOR_UNLOAD_ON_FIELD, true)
+        end
+    elseif self.state == self.states.UNLOADING_ON_FIELD then
+        -- Unloading
+        self:driveUnloadOnField()
+        self:callUnloaderWhenNeeded()
+
+    end
+    return AIDriveStrategyFieldWorkCourse.getDriveData(self, dt, vX, vY, vZ)
+end
+
+--- We need to update pipeoffset after turn as fruit side may have changed
+function AIDriveStrategyChopperCourse:resumeFieldworkAfterTurn(ix)
+    self:updatePipeOffset(ix)
+    AIDriveStrategyChopperCourse.superClass().resumeFieldworkAfterTurn(self, ix)
+end
+
+-----------------------------------------------------------------------------------------------------------------------
+--- Event listeners
+-----------------------------------------------------------------------------------------------------------------------
+function AIDriveStrategyChopperCourse:onWaypointPassed(ix, course)
+    if self.state == self.states.UNLOADING_ON_FIELD and
+            (self.unloadState == self.states.DRIVING_TO_SELF_UNLOAD or
+                    self.unloadState == self.states.DRIVING_TO_SELF_UNLOAD_AFTER_FIELDWORK_ENDED or
+                    self.unloadState == self.states.RETURNING_FROM_SELF_UNLOAD) then
+        -- nothing to do while driving to unload and back
+        return AIDriveStrategyFieldWorkCourse.onWaypointPassed(self, ix, course)
+    end
+
+    -- Is this really need any more seeing how fruit check is handled by updatePipeOffset
+    self:checkFruit()
+
+    if self.state == self.states.WORKING then
+        self:estimateDistanceUntilFull(ix)
+        self:callUnloaderWhenNeeded()
+        -- Alteration from parent function we need to check every once in a while to see if the next unloader incoming is ready to replace out current unloader
+        self:checkNextUnloader()
+    end
+
+    AIDriveStrategyFieldWorkCourse.onWaypointPassed(self, ix, course)
+end
+
+function AIDriveStrategyChopperCourse:checkRendezvous()

```

Comment:
Needs explanation here, what a rendezvous means here, as obviously, with a chopper, an unloader always must be present, not like with a combine which can arrange to meet an unloader in the future